File: 프로그래머스/0/120813. 짝수는 싫어요/짝수는 싫어요.java
class Solution {
    public int[] solution(int n) {
        int size = n % 2 == 0 ? n / 2 : n / 2 + 1;
        int[] arr = new int[size];
        
        // int cur = 1;
        // for (int i=0; i<arr.length; i++) {
        //     arr[i] = cur;
        //     cur += 2;
        // }
        
        int i = 0;
        int cur = 1;
        while (cur <= n) {
            arr[i++] = cur;
            cur += 2;
        }
        
        return arr;
    }
}

File: 프로그래머스/0/120813. 짝수는 싫어요/짝수는 싫어요.py
def solution(n):
    answer = []  # 홀수를 저장할 리스트

    for i in range(1, n+1, 2):  # 1부터 n까지 2씩 증가 (홀수만 선택)
        answer.append(i)  # 홀수를 리스트에 추가

    return answer  # 생성된 홀수 리스트 반환

# 테스트 예제 실행
print(solution(10))  # [1, 3, 5, 7, 9]
print(solution(15))  # [1, 3, 5, 7, 9, 11, 13, 15]